### DISCUSSION TITLE
Where is setJointMotorControl2 defined?

### DISCUSSION THREAD

**Discussion by sash-a** [bulletphysics/bullet3#3915] [Q&A]
Hi there, I'm currently trying to extend a bullet API for julia and I can't seem to find where `setJointMotorControl2` is defined. I see [`setJointMotorControl` here](https://github.com/bulletphysics/bullet3/blob/cdd56e46411527772711da5357c856a90ad9ea67/examples/SharedMemory/b3RobotSimulatorClientAPI_NoDirect.cpp#L762), but looking at python code `setJointMotorControl2` seems to have more args than `setJointMotorControl` so I'm assuming it's a different method. For instance in robot_bases.py:

```
def disable_motor(self):
  self._p.setJointMotorControl2(self.bodies[self.bodyIndex],
                              self.jointIndex,
                              controlMode=pybullet.POSITION_CONTROL,
                              targetPosition=0,
                              targetVelocity=0,
                              positionGain=0.1,
                              velocityGain=0.1,
                              force=0)
```

However `setJointMotorControl` uses `b3RobotSimulatorJointMotorArgs` which do not contain `positionGain`, `velocityGain` or `force` fields. If someone could explain to me the different between the two methods or point me towards the implementation of `setJointMotorControl2` that would be appreciated as I can't seem to find it in this repo.

**Comment by BinglunWang** [comment]
Hi sash-a,

I got the same question. 

And do you know how exactly setJointMotorControl2 works for POSITION_CONTROL?

I hope to find out how they get the desired PD control values.

**Reply by sash-a** [reply]
Hi sorry this was a long time ago and so I don't remember. I would recommend using MuJoCo as I remember I had more success with that.
